src/libsyntax/tokenstream.rs

```diff
@@ -368,8 +368,21 @@ impl TokenStream {
     // This is otherwise the same as `eq_unspanned`, only recursing with a
     // different method.
     pub fn probably_equal_for_proc_macro(&self, other: &TokenStream) -> bool {
-        let mut t1 = self.trees();
-        let mut t2 = other.trees();
+        fn semantic_tree(tree: &TokenTree) -> bool {
+            match tree {
+                | TokenTree::Token(_, Token::Comma)
```

Comment:
Could you throw a comment here for why we're ignoring all comma tokens?

One case this may get tripped up is `(x)` vs `(x,)`, though :(

Reply:
This appears to be the most commonly hit span-loss in practice (see https://github.com/SergioBenitez/Rocket/issues/811, my own comment in https://github.com/rust-lang/rust/issues/43081#issuecomment-409078027, and https://github.com/dtolnay/syn/issues/482). In general, the pretty printer inserts trailing commas everywhere, and users don't. So all three of the following lose span information:

```rust
struct Foo { a: usize }
let x = Foo { a: 10 };
match x { Foo(..) => 10 }
```